instrumentation/github.com/gin-gonic/gin/otelgin/option.go

```diff
@@ -30,9 +31,9 @@ type config struct {
 	SpanNameFormatter SpanNameFormatter
 }
 
-// Filter is a predicate used to determine whether a given http.request should
+// Filter is a predicate used to determine whether a given gin.Context should
 // be traced. A Filter must return true if the request should be traced.
-type Filter func(*http.Request) bool
+type Filter func(*gin.Context) bool
```

Comment:
This is a breaking change. 

I would rather add `type GinFilter func(*gin.Context) bool` and `WithGinFilter` and keep the old functionality. Reasons:

1. users already happy with existing API would not need to change anything
2. some may already have some code designed/reusable for `net/http`

I am letting @hanyuancheung (who is the codeowner) whether it is a good idea or not.